### PR TITLE
feat(#5579): add topics to reports

### LINKED ISSUE #3028: Convert download case contact notes from CSV to Excel and turn on auto-wrap
**What type(s) of user does this feature affect?**
 - volunteers?
 - supervisors?
 - admins?
 - all casa admins?

**Description**

Case supes asking, when download conacct notes to print,
wondering if “note fields” could wrap text automatically, don’t want to have to wrap ourselves

if its not a excel (and its probably not) we can convert it

**Screenshots of current behavior, if any**  
You can paste images on the clipboard here

_QA Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Wednesday 6-8pm Pacific), stakeholder news, and upcoming new issues.

### LINKED ISSUE #5579: Migrate Case Contact Notes to Corresponding Section When Generating Court Report
## What type(s) of user does this feature affect?
 - volunteers
 - supervisors
 - admins

## Description
When a user generates a court report, they should be able to pick the date range for case contact notes to flow into their corresponding section of the court report doc. 

[Figma File](https://www.figma.com/file/OuJMUdTiBbyNhiiOtqVaBG/Generate-Report-Flow?type=design&node-id=0%3A1&mode=design&t=cGQDpVrDhC5EXrMv-1)
[Prototype](https://www.figma.com/proto/OuJMUdTiBbyNhiiOtqVaBG/Generate-Report-Flow?page-id=0%3A1&type=design&node-id=35-6399&viewport=287%2C1263%2C0.54&t=JDPdyDjDKpJkdcr5-1&scaling=min-zoom&starting-point-node-id=35%3A6399)

The prototype shows this page - Please implement here:
image

Please also implement the modal behavior where users can generate a court report from "Case Details" using the "Generate Report" button on the top right of the screen. The case should default to whatever case they are in. 
image
image

## How to access the QA site
_Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin
- all_casa_admin1@example.com view site as an all casa admin
  - go to `/all_casa_admins/sign_in`  

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in [slack](https://join.slack.com/t/rubyforgood/shared_invite/zt-21pyz2ab8-H6JgQfGGI0Ab6MfNOZRIQA) #casa channel to ask questions quickly. And [discord](https://discord.gg/qJcw2RZH8Q) for office hours (currently Tuesday 5-7pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by elasticspoon**
Fixes https://github.com/rubyforgood/casa/issues/5579
Fixes #3028
Depends on
- [x] #5590
- [x] #5593

## Overview
In #5388 I added` ContactTopics` and `ContactTopicAnswers` to `CaseContacts`. This allowed `CasaOrg`s to set topics and Users to create `ContactTopicAnswers` on specific `CaseContacts`. 

Now we want to have those answers appear in the court reports.

This PR modifies `CourtReportContext` to include topic information, as well as modifies `CourtReport` templates to accept that information.

## Process

Topics are taken out of the database and massaged into the following shape:

```rb
  [
    {topic: "Question 1", details: "Details 1", answers: [
      {date: "12/02/20", medium: "Type A1, Type B1", value: "Answer 1"},
      {date: "12/03/20", medium: "Type A2, Type B2", value: "Answer 3"}
    ]},
    {topic: "Question 2", details: "Details 2", answers: [
      {date: "12/02/20", medium: "Type A1, Type B1", value: "Answer 2"},
      {date: "12/04/20", medium: "Type A3, Type B3", value: "Answer 5"}
    ]},
    {topic: "Question 3", details: "Details 3", answers: [
      {date: "12/03/20", medium: "Type A2, Type B2", value: "No Answer Provided"},
      {date: "12/04/20", medium: "Type A3, Type B3", value: "No Answer Provided"}
    ]}
  ]

```

That info gets filled in via document templates like so

image

Just to label the parts:
image

1. Contact Topic
2. Topic details
3. Medium of contact
4. Contact Date
5. Answer to topic on that contact

### CODE REVIEW

**Comment by compwron** [comment]
Fix merge conflict
```
This branch has conflicts that must be resolved
Use the [web editor](https://github.com/rubyforgood/casa/pull/5601/conflicts) or the to resolve conflicts.
Conflicting files
app/views/case_contacts/form/expenses.html.erb
```

**Comment by compwron** [comment]
Nice labeled diagram :)

File: app/documents/templates/prince_george_report_template.docx
Comment (compwron) [inline app/documents/templates/prince_george_report_template.docx:1]: Why did this one get smaller? Seems strange and maybe not good
Screenshot 2024-05-14 at 8 50 24 AM

File: app/models/case_court_report_context.rb [hunk 1]
```diff
@@ -88,6 +89,51 @@ def org_address(path_to_template)
     @volunteer.casa_org.address if @volunteer && is_default_template
   end
 
+  # Sample output
```
Comment (compwron) [inline app/models/case_court_report_context.rb:92]: good comment :)

File: app/models/case_court_report_context.rb [hunk 2]
```diff
@@ -88,6 +89,51 @@ def org_address(path_to_template)
...
+  #   {date: "12/02/20", medium: "Type A1, Type B1", value: "Answer 2"},
+  #   {date: "12/04/20", medium: "Type A3, Type B3", value: "Answer 5"}
+  # ]},
+  # "Question 3" => {topic: "Question 3", details: "Details 3", answers: [
+  #   {date: "12/03/20", medium: "Type A2, Type B2", value: "No Answer Provided"},
+  #   {date: "12/04/20", medium: "Type A3, Type B3", value: "No Answer Provided"}
+  # ]}
+  # }
+  def court_topics
+    topics = ContactTopic
```
Comment (compwron) [inline app/models/case_court_report_context.rb:109]: nice query. It could be slow but we are working with small numbers I think we're fine

**Review by compwron — APPROVED** [review]
good tests :)

File: app/documents/templates/prince_george_report_template.docx
Comment (elasticspoon) [reply app/documents/templates/prince_george_report_template.docx:1]: No clue. The template got longer so that's weird.
old: image

new: image

**Review by compwron — APPROVED** [review]
Heck yes

**Pull request merged**